### PR TITLE
Adding HTTPS options for WebHDFS

### PR DESCRIPTION
**Pull request by alericmckearn**
Currently, Hue is hard coded to http for webhdfs (return "http://%s:%s/webhdfs/v1" % (host, port)).  This PR makes it dynamically pull SSL from hdfs-site settings with the ability to override the port from hue.ini

Note -> i just moved the port into the hdfs-site.py so i need to test that and make sure it works since previously i was injecting it in webhdfs.py (this seemed cleaner though it does seem to break the convention of no hue.ini settings in hdfs-site.py).

### CODE REVIEW

File: desktop/libs/hadoop/src/hadoop/hdfs_site.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 import conf
 import confparse
-
+import hadoop.conf
```
Comment (romainr) [inline desktop/libs/hadoop/src/hadoop/hdfs_site.py]: Is is used?

**Comment by alericmckearn** [comment]
Updated per comments

File: desktop/libs/hadoop/src/hadoop/hdfs_site.py [hunk 2]
```diff
@@ -51,6 +52,12 @@ def get_umask_mode():
 
   return int(umask, 8)
 
+def get_webhdfs_ssl():
+  settings = {"protocol": "http", "port": hadoop.conf.DEFAULT_NN_HTTP_PORT}
+  if get_conf().get(_CNF_HTTP_POLICY, 'http') == "HTTPS_ONLY":
+    settings["protocol"] = 'https'
+    settings["port"] = get_conf().get(_CNF_WEBHDFS_HTTPS_PORT, hadoop.conf.DEFAULT_NN_HTTP_PORT)
```
Comment (romainr) [inline desktop/libs/hadoop/src/hadoop/hdfs_site.py]: Should be just DEFAULT_NN_HTTP_PORT but I will just fix it with a commit on top

**Comment by romainr** [comment]
https://review.cloudera.org/r/13779/

**Pull request merged**